src/main/java/frc/robot/Auto.java

```diff
@@ -0,0 +1,247 @@
+	 * @return A runnable command
+	 */
+	public static ClimbPad climbPad(DriveBase db, Gyro pitch, double evel, double ivel) {
+		return new ClimbPad(db, pitch, evel, ivel);
+	}
+	/**	Get a command for parking on the charging pad based on the gyro angle
+	 * @param db The drivebase subsystem
+	 * @param pitch A gyro implementation for the robot's pitch axis
+	 * @param kp The p-gain value - VOLTS PER DEGREE [of error]
+	 * @return The command
+	 */
+	public static BalancePark balancePark(DriveBase db, Gyro pitch, double kp) {
+		return new BalancePark(db, pitch, kp);
+	}
+
+
+
+
+    public static class ActivePark extends CommandBase {
+
+        private final DriveBase drivebase;
+        private final double volts_per_meter;
+        private double linit, rinit;
+
+        public ActivePark(DriveBase db, double p) { // p is the proportional gain, in volts per meter [error]
+            this.drivebase = db;
+            this.volts_per_meter = p;
+        }
+
+        @Override
+        public void initialize() {
+            this.linit = this.drivebase.getLeftPosition();
+            this.rinit = this.drivebase.getRightPosition();
+        }
+        @Override
+        public void execute() {
+            double le = this.drivebase.getLeftPosition() - this.linit;
+            double re = this.drivebase.getRightPosition() - this.rinit;
+            this.drivebase.setDriveVoltage(
+                -le * this.volts_per_meter,     // we are assuming that positive position for the encoders is the same direction as positive voltage
+                -re * this.volts_per_meter
+            );
+        }
+        @Override
+        public void end(boolean interrupted) {
+            this.drivebase.setDriveVoltage(0.0, 0.0);
+        }
+        @Override
+        public boolean isFinished() {
+            return false;
+        }
+
+    }
+    public static class BalancePark extends CommandBase {
+
+		private final DriveBase drivebase;
+		private final Gyro pitch_axis;
+		private final double volts_per_degree;
+		private double
+			left_target, right_target,
+			pitch_target;
+
+		public BalancePark(DriveBase db, Gyro pitch, double kp_v) {
+			this.drivebase = db;
+			this.pitch_axis = pitch;
+			this.volts_per_degree = kp_v;
+		}
+
+		@Override
+		public void initialize() {
+			this.left_target = this.drivebase.getLeftPosition();
+			this.right_target = this.drivebase.getRightPosition();
+			this.pitch_target = this.pitch_axis.getAngle();
+		}
+		@Override
+		public void execute() {
+			double ae = this.pitch_target - this.pitch_axis.getAngle();
+			this.drivebase.setDriveVoltage(
+				ae * this.volts_per_degree,
+				ae * this.volts_per_degree
+			);
+		}
+		@Override
+		public void end(boolean i) {
+			this.drivebase.setDriveVoltage(0, 0);
+		}
+		@Override
+		public boolean isFinished() {
+			return false;
+		}
+
+	}
+	public static class ClimbPad extends CommandBase {
+
+        public static final double
```

Comment:
Is there a reason why these instance variables are declared public?

Reply:
Those are static which means that they aren't instance variables - so yes there is a reason and that is because since they are final and meant to be constant values, I wanted them to be accessible outside the class if needed.

Reply:
Ok, I missed that they are static; however, just from looking at them, it seems like these constants only make sense in the context of the implementation of the ClimbPad class.  Generally it's better to expose as little of the class's implementation as possible to other classes so that you can make changes to the implementation of the class without breaking other code that is using that class.

If you really need to access the constants from other classes, wouldn't it be better for them to be in the Constants class?

Reply:
Yep, I just didn't want to add clutter if these values are only temporary (I'm not done with this routine). Once it is done they I will add the constants to the actual Constants.java file.

Reply:
Its also just so I can see what those values are easily and know for sure what they are for - if they were in constants I might forget if I was lazy with naming them.